daemon/src/position_metrics.rs

```diff
@@ -168,10 +168,20 @@ impl Cfd {
                 is_open: true,
                 ..self
             },
-            ContractSetupFailed => Self {
-                is_failed: true,
-                ..self
-            },
+            ContractSetupFailed => {
+                // This is needed because we have a CFD (bd6221b0-47f6-4b14-ac9b-04cfa355cb9b) in
+                // the database where `OfferRejected` and `ContractSetupFailed` was appended. This
+                // was due to a bug that has since been fixed; `OfferRejected` and
+                // `ContractSetupFailed` are mutually exclusive.
+                if self.is_rejected {
```

Comment:
What do you think of logging something here? I understand that *this should never happen*, but if it ever happen, it should mean something is wrong.

Reply:
I'll insist on my earlier point: I think this would be simpler if we just have a `CfdState` enum with all the variants and you don't have to reason about what was there before. We simply overwrite it with the state that corresponds to the last event we process.

Reply:
> What do you think of logging something here? I understand that this should never happen, but if it ever happen, it should mean something is wrong.

I think this specific case is actually not a big deal. Even if it happens - both of them are treated as failed internally. Since we know that we have one old position that has this issue I would not log anything for now. We already have checks for this in place when we move CFDs into closed / failed - if we want to log I'd do it there.